diff --git a/pkg/cache/v3/delta_test.go b/pkg/cache/v3/delta_test.go
index 73164328..308b1154 100644
--- a/pkg/cache/v3/delta_test.go
+++ b/pkg/cache/v3/delta_test.go
@@ -3,6 +3,7 @@ package cache_test
 import (
        "fmt"
        "reflect"
+       "sync"
        "testing"
        "time"

@@ -232,6 +233,11 @@ func BenchmarkDeltaSnapshotCache(b *testing.B) {

        for _, typ := range testTypes {
                b.Run(typ, func(b *testing.B) {
+                       wg := sync.WaitGroup{}
+                       wg.Add(2)
+                       for i := 0; i < 2; i++ {
+                               go func() {
+                                       defer wg.Done()
                        for n := 0; n < b.N; n++ {
                                value, _ := c.CreateDeltaWatch(&discovery.DeltaDiscoveryRequest{
                                        Node: &core.Node{
@@ -247,6 +253,9 @@ func BenchmarkDeltaSnapshotCache(b *testing.B) {
                                        b.Error("failed to receive a snapshot response")
                                }
                        }
+               }()
+       }
+       wg.Wait()
                })
        }
 }
